prov/tcp/src/tcpx.h

```diff
@@ -149,11 +150,17 @@ struct tcpx_cq_data_hdr {
 			 TCPX_IOV_LIMIT +			\
 			 TCPX_MAX_INJECT_SZ)
 
+typedef void (*tcpx_hdr_ntoh_fn_t)(struct tcpx_base_hdr *hdr);
+typedef void (*tcpx_hdr_hton_fn_t)(struct tcpx_base_hdr *hdr);
+typedef uint64_t (*tcpx_hdr_sz_ntoh_fn_t)(struct tcpx_base_hdr *hdr);
+typedef uint16_t (*tcpx_hdr_flgs_ntoh_fn_t)(struct tcpx_base_hdr *hdr);
+
 struct tcpx_rx_detect {
 	union {
 		struct tcpx_base_hdr	base_hdr;
 		uint8_t		       	max_hdr[TCPX_MAX_HDR_SZ];
 	} hdr;
+	tcpx_hdr_ntoh_fn_t	hdr_to_host_order;
```

Comment:
I don't understand why this is needed.  It seems that associating the conversion handlers as part of the ep structure would be sufficient.

Reply:
This pointer is added to make sure that we convert the received header from network to host order only once. New rx_entry allocation may fail and result in calling the ntoh multiple times. So I just calling this function via this ptr for first time and making it null after that.